tokenizers/src/tokenizer/mod.rs

```diff
@@ -795,14 +795,21 @@ where
     }
 
     /// Decode the given ids, back to a String
-    pub fn decode(&self, ids: Vec<u32>, skip_special_tokens: bool) -> Result<String> {
+    pub fn decode(
+        &self,
+        ids: Vec<u32>,
+        skip_special_tokens: bool,
+        spaces_between_special_tokens: bool,
+    ) -> Result<String> {
         let tokens = ids
             .into_iter()
             .filter_map(|id| {
                 self.added_vocabulary
                     .id_to_token(id, &self.model)
                     .filter(|token| {
-                        !skip_special_tokens || !self.added_vocabulary.is_special_token(token)
+                        !skip_special_tokens
+                            || !self.added_vocabulary.is_special_token(token)
+                            || !spaces_between_special_tokens
```

Comment:
Huh ?

This seems a bit odd.  spaces_between_special_tokens should control how to reconstruct the string no ?

Reply:
Yep, I think I am missing this part